code/modules/holomaps/holomap_datum.dm

```diff
@@ -0,0 +1,185 @@
+	var/image/holomap_base
+	var/static/image/default_holomap
+
+/datum/holomap_interface/New(obj/item/holder)
+	src.holder = holder
+	..()
+
+/datum/holomap_interface/Destroy()
+	QDEL_NULL(holder)
+	return ..()
+
+/datum/holomap_interface/process()
+	update_holomap()
+
+/datum/holomap_interface/proc/draw_special()
+	return
+
+/datum/holomap_interface/proc/draw_special_icon(var/filter, var/obj/holomap_holder)
+	var/image/indicator = image('icons/holomap_markers.dmi', (filter))
+	indicator.plane = ABOVE_HUD_PLANE
+	indicator.layer = ABOVE_HUD_LAYER
+	indicator.loc = activator.hud_used.holomap_obj
+	var/turf/location = get_turf(holomap_holder)
+	if(!(location.z == 1))
+		indicator.icon_state = "error"
+	if(holomap_holder == holder)
+		indicator.icon_state = "you"
+	if(indicator.icon_state == "error")
+		indicator.pixel_x = (rand(6,9)) * PIXEL_MULTIPLIER
+		indicator.pixel_y = (rand(6,9)) * PIXEL_MULTIPLIER
+	indicator.pixel_x = (location.x - 6) * PIXEL_MULTIPLIER
+	indicator.pixel_y = (location.y - 6) * PIXEL_MULTIPLIER
+	holomap_images += indicator
+
+/datum/holomap_interface/proc/activate(mob/user, var/role_filter)
+	if(activator)
+		return
+	activator = user
+	if(!default_holomap)
+		default_holomap = image(generateHoloMap())
+	holomap_base = default_holomap
+	holomap_base.layer = HUD_LAYER
+	holomap_base.plane = HUD_PLANE
+	switch(role_filter)
+		if("nuclear")
+			holomap_base.color = HOLOMAP_NUCLEAR_COLOR
+		if("deathsquad")
+			holomap_base.color = HOLOMAP_DEATHSQUAD_COLOR
+		if("vox")
+			holomap_base.color = HOLOMAP_VOX_COLOR
+		if("ert")
+			holomap_base.color = HOLOMAP_ERT_COLOR
+	activator.hud_used.holomap_obj.overlays += holomap_base
+	START_PROCESSING(SSobj, src)
+
+/datum/holomap_interface/proc/deactivate_holomap()
+	STOP_PROCESSING(SSobj, src)
+	if(!activator)
+		return
+	activator.hud_used.holomap_obj.overlays -= holomap_base
+	if(activator.client)
+		activator.client.images -= holomap_images
+	for(var/i in holomap_images)
+		qdel(i)
+	holomap_images.Cut()
+	activator = null
+
+/datum/holomap_interface/proc/update_holomap()
+	if(!activator || !activator.client)
+		deactivate_holomap()
+		return
+
+	if(length(holomap_images))
+		activator.client.images -= holomap_images
+		for(var/i in holomap_images)
+			qdel(i)
+		holomap_images.Cut()
+
+	draw_special()
+
+	activator.client.images |= holomap_images
+
+/datum/holomap_interface/deathsquad/draw_special()
+	for(var/obj/item/clothing/head/helmet/space/deathsquad/D in deathsquad_helmets)
+		draw_special_icon_mob("deathsquad", D)
+
+/datum/holomap_interface/nuclear/draw_special()
+	for(var/obj/item/clothing/head/helmet/space/rig/syndi/S in nuclear_holo)
+		draw_special_icon_mob("nuclear", S)
+	for(var/mob/living/silicon/robot/syndicate/R in nuclear_holo)
+		draw_special_icon_robot("syborg", R)
+	for(var/obj/machinery/computer/syndicate_station/SS in nuclear_holo)
+		draw_shuttle_icon("syndishuttle", SS)
+
+/datum/holomap_interface/proc/draw_shuttle_icon(var/filter, var/shuttle_loc)
```

Comment:
https://github.com/TauCetiStation/TauCetiClassic/wiki/Code-Convention#%D0%9E%D0%B1%D1%8A%D1%8F%D0%B2%D0%BB%D0%B5%D0%BD%D0%B8%D0%B5-%D0%B0%D1%80%D0%B3%D1%83%D0%BC%D0%B5%D0%BD%D1%82%D0%BE%D0%B2-%D0%B2-%D0%BF%D1%80%D0%BE%D1%86%D0%B5%D0%B4%D1%83%D1%80%D0%B0%D1%85